src/audio_decoder_midi.cpp

```diff
@@ -378,6 +402,10 @@ void AudioDecoderMidi::meta_event(int event, const void* data, std::size_t size)
 
 void AudioDecoderMidi::reset() {
 	// MIDI reset event
+	if (mididec->NeedsSoftReset()) {
+		// SoundOff every channel: necessary for synths like macOS which tend to get stuck
+		SendMessageToAllChannels(midimsg_all_sound_off(0));
```

Comment:
sorry have to ask you for another small change:

Can you move this new code block between the GM reset and the pitch bend block? (I hope this doesn't break macOS?)

Tested this again in fluidsynth and the all sound off will not work properly when executed before the GM reset (same audio glitch remains).

```cpp
        // GM system on (resets most parameters)
        const unsigned char gm_reset[] = { 0xF0, 0x7E, 0x7F, 0x09, 0x01, 0xF7 };
        mididec->SendSysExMessage(gm_reset, sizeof(gm_reset));

        // MIDI reset event
        if (mididec->NeedsSoftReset()) {
                // SoundOff every channel: necessary for synths like macOS which tend to g>
                SendMessageToAllChannels(midimsg_all_sound_off(0));
        }

        // Set the Pitch bend range to 256
        for (int channel = 0; channel < 16; channel++) {
```